providers/in-process-flagd/README.md

```diff
@@ -0,0 +1,69 @@
+# In-process Flagd Provider
+
+This provider implements parts of the functionality of [Flagd](https://github.com/open-feature/flagd), and follows its
+specification for [in-process providers](https://github.com/open-feature/flagd/blob/main/docs/other_resources/in-process-providers/specification.md).
+
+## Setup
+To use flagd with the [OpenFeature SDK](https://github.com/open-feature/go-sdk) set the provider to the `openfeature` global singleton as shown below (using default values which align with those of `flagd`)
+```go
+openfeature.SetProvider(inprocessflagd.NewProvider())
```

Comment:
Naming. we can go for either `inprocessflagd` or `flagdInProcess` 🤔

Any opinion @thisthat @toddbaert  @mowies @odubajDT  ?

Reply:
I favor the first one: `inprocessflagd`, since it reads better. 
Some reading for guidance on pkg naming: https://go.dev/blog/package-names